streamalert_cli/terraform/generate.py

```diff
@@ -363,8 +363,13 @@ def generate_cluster(config, cluster_name):
 
 
 def cleanup_old_tf_files(config):
-    """Cleanup old .tf files, these are now .tf.json files per Hashicorp best practices"""
-    files_for_removal = set(config.clusters()).union({'athena', 'main'})
+    """
+    Cleanup old .tf files, these are now .tf.json files per Hashicorp best practices
+
+    These are the files left behind
+    ['rules_engine.tf.json', 'alert_processor.tf.json', 'alert_merger.tf.json']
+    """
+    files_for_removal = set(config.clusters()).union({'athena', 'main', 'metric_filters'})
```

Comment:
I'll be honest - I don't know why we aren't just removing _all_ `*.tf.json` files here. do you know @chunyong-lin ?

Reply:
this might be an oversight or something that was never changed when we started moving to more tf.json config files.

Reply:
I also struggle to understand these few lines.

Reply:
So my guess is that when everything was `*.tf`, you wanted to preserve the provider.tf which was being tracked by `git`.

Now that every important terraform file produced is `*.tf.json` we could remove the if statement and see if that works :)?

Reply:
yeah - I think I support this and anything that ends in `.tf.json` should be removed. the `providers.tf` file is fairly new (last few months) so that wouldn't explain this

Reply:
consider it shipped!

I've removed the code that wasn't required anymore as well

Reply:
@chunyong-lin I think this is good for me - I'll let you have final say!

Reply:
It makes lot of sense! Found the history about `*.tf` vs. `*.tf.json` transition. The reason we are not just removing all `*.tf.json` files was because we used `*.tf` to store terraform configuration files before 2018. During that time, we need to keep some internal/customized `*.tf` file(s). 
Then we switched to use `*.tf.json` files... 
TL;DR, few interesting commit history
* https://github.com/airbnb/streamalert/pull/949/files
* https://github.com/airbnb/streamalert/commit/64611e8865875d35b4c3263a0afd525cc1078be5
* https://github.com/airbnb/streamalert/commit/96fd3c3bdd34dbc1a532e4c4f6b768ffe98a25cb